expression/builtin_string.go

```diff
@@ -1061,49 +1061,54 @@ type unhexFunctionClass struct {
 }
 
 func (c *unhexFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinUnHexSig{newBaseBuiltinFunc(args, ctx)}
+	var retFlen int
+
+	argType := args[0].GetType()
+	switch t := args[0].GetTypeClass(); t {
+	case types.ClassString:
+		// Use UTF-8 as default charset, so there're (Flen * 3 + 1) / 2 byte-pairs
+		retFlen = (argType.Flen*3 + 1) / 2
+
+	case types.ClassInt, types.ClassReal, types.ClassDecimal:
+		// For number value, there're (Flen + 1) / 2 byte-pairs
+		retFlen = (argType.Flen + 1) / 2
+
+	default:
+		return nil, errors.Errorf("Unhex invalid args, need int or string but get %T", t)
+	}
+
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf.tp.Flen = retFlen
+	types.SetBinChsClnFlag(bf.tp)
+	sig := &builtinUnHexSig{baseStringBuiltinFunc{bf}}
 	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
```

Comment:
Sorry, I didn't get the idea. Did you mean that add 
`err := c.verifyArgs(args)`
`if err != nil { return nil, errors.Trace(err) }`
at line 1064